### ISSUE TITLE
Allow Stores instantiation from non-main threads

### ISSUE THREAD

**Issue by arkivanov** [badoo/MVIKotlin#189]
Jetpack Compose will have multi-threaded compositions at some point, and so it should be possible to instantiate `Stores` from non-main threads. The `Bootstrapper` is currently invoked from `Store` constructor, it should continue be executing on main thread. We probably need an ability to manually trigger the `Bootstrapper`.

**Comment by arkivanov** [comment]
Turned out this is not required. We can use `Lazy` and bootstrap `Stores` manually:

```kotlin
@Composable
fun Some() {
    val store: SomeStore by rememberLazyStore { SomeStoreFactory(/*...*/).create() }
}

@Composable
fun <T : Store<*, *, *>> rememberLazyStore(initializer: () -> T): Lazy<T> {
    val storeLazy = remember(initializer) { lazy(initializer) }

    DisposableEffect(initializer) {
        storeLazy.value.let {} // Bootstrap the store
        onDispose { storeLazy.value.dispose() }
    }
    
    return storeLazy
}
```

**Comment by arkivanov** [comment]
Will be delivered in version 2.1.0

**Comment by arkivanov** [comment]
Closed via #217
